source/core/src/main/com/csse3200/game/components/maingame/MainGameDisplay.java

```diff
@@ -234,21 +258,22 @@ public void clicked(InputEvent event, float x, float y) {
                         TowerType selected = ServiceLocator.getCurrencyService().getTower();
                         if (selected == towers.get(4)) {
                             ServiceLocator.getCurrencyService().setTowerType(null);
-                            tower5.setChecked(false);
+
+                            towerToggle(tower5);
```

Comment:
Was this meant to be towerToggle(null)? It breaks the previous pattern I've seen

Reply:
good spotting! This was indeed supposed to be towerToggle(null)